submissions/ik-web/frogger/app.js

```diff
@@ -0,0 +1,202 @@
+    if (this.x - borderCollision <= this.player.x        // Left
+        && this.y - borderCollision <= this.player.y     // Up
+        && this.x + borderCollision >= this.player.x     // Right
+        && this.y + borderCollision >= this.player.y) {  // Down
+        this.player.x = playerStartPosX;
+        this.player.y = playerStartPosY;
+
+        setTimeout(() => {
+            alert(`Game over...\nYou received ${this.player.level} level.`);
+            resetGame();
+        }, 100);
+    }
+};
+
+// Method for enemies update
+Enemy.prototype.update = function(dt) {
+    this.enemyCollision();
+
+    if (this.x > areaWidth) {
+        this.x = enemyPosX;
+        this.speed = enemySpeed();
+    } else {
+        this.x += this.speed * dt;
+    }
+};
+
+//----------------------------------------------
+// Constructor for create player
+const Player = function(x, y, sprite) {
+    Character.call(this, x, y, sprite);
+    this.level = playerStartLevel;
+};
+
+Player.prototype = Object.create(Character.prototype);
+
+//Method for checking game progress
+Player.prototype.checkProgress = function() {
+    if (this.y < 0) {
+
+        setTimeout(() => {
+            this.x = playerStartPosX;
+            this.y = playerStartPosY;
+            this.level++;
+            setLevel(this.level);
+            addEnemy();
+
+            if (this.level === levelForWin) {
+                alert('Victory!!!');
+                resetGame();            
+            } else {
+                alert(`Level UP!\nYour level is ${this.level}`);
+            }
+
+        }, 100);
+    }
+};
+
+// Player controls
+Player.prototype.handleInput = function(btn) {
+    if (btn === 'left') {
+        if (this.x > indentBorderLeft) {
+            this.x -= playerStepX;
+        }
+    } else if (btn === 'up') {
+        if (this.y > indentBorderTop) {
+            this.y -= playerStepY;
+
+            this.checkProgress(); 
+        }
+    } else if (btn === 'right') {
+        if (this.x < indentBorderRight) {
+            this.x += playerStepX;
+        }
+    } else if (btn === 'down') {
+        if (this.y < indentBorderBottom) { 
+            this.y += playerStepY;
+        }
+    }
+};
+
+// ---------------------------------------------
+// This listens for key presses and sends the keys to Player.handleInput() method.
+document.addEventListener('keyup', function(e) {
+    var allowedKeys = {
+        37: 'left',
+        38: 'up',
+        39: 'right',
+        40: 'down'
+    };
+
+    player.handleInput(allowedKeys[e.keyCode]);
+});
+// ---------------------------------------------
+// ---------------------------------------------
+// ---------------------------------------------
+// Create player character
+const player = new Player(playerStartPosX, playerStartPosY, playerSprite);
+
+// Create first enemy
+let enemy = new Enemy(enemyPosX, enemyPosY(), enemySprite, enemySpeed(), player);
+let allEnemies = [enemy];
```

Comment:
```suggestion
let allEnemies = [];
addEnemy();
```
This way the approach to create enemies is consistent with the one you already use and also makes code shorter.